### PR TITLE
Bump SDK version

### PR DESCRIPTION
**Pull request by ValarDragon**
Closes: #XXX

## Description

Big thing is this improves SDK pruning=default settings, and corrects comments. It also has fixes that help on main. 

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by sunnya97** [comment]
Make sure to update the comment in the go.mod

File: go.mod [hunk 1]
```diff
@@ -121,7 +121,7 @@ require (
 
 replace (
 	// Our cosmos-sdk branch is:  https://github.com/osmosis-labs/cosmos-sdk  v0.44.3x-osmo-v5
```
Comment (sunnya97) [inline go.mod:123]: Make sure to update this commend

Comment (ValarDragon) [reply go.mod:123]: That comment is correct still.

Comment (sunnya97) [reply go.mod:123]: We're using a branch instead of a release?

**Pull request merged**